src/query/api/v1/handler/prometheus/native/read_instantaneous.go

```diff
@@ -87,13 +87,12 @@ func (h *PromReadInstantHandler) ServeHTTP(w http.ResponseWriter, r *http.Reques
 		return
 	}
 
-	engineOpts := &executor.EngineOptions{
-		QueryContextOptions: models.QueryContextOptions{
-			LimitMaxTimeseries: fetchOpts.Limit,
-		},
-	}
+	engineOpts := h.engine.Opts().SetQueryContextOptions(models.QueryContextOptions{
+		LimitMaxTimeseries: fetchOpts.Limit,
+	})
+	h.engine = h.engine.SetOpts(engineOpts)
```

Comment:
Again, this feels racy

Reply:
Yes this will race per request, need to avoid that.